eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway/DoctrineDatabase.php

```diff
@@ -1310,6 +1310,93 @@ public function deleteUrlAliasesWithBrokenLink()
         return $query->execute();
     }
 
+    /**
+     * Attempt repairing data corruption for broken archived URL aliases for Location,
+     * assuming there exists restored original (current) entry.
+     *
+     * @param int $locationId
+     */
+    public function repairBrokenUrlAliasesForLocation($locationId)
+    {
+        $urlAliasesData = $this->getUrlAliasesForLocation($locationId);
+
+        $originalUrlAliases = $this->filterOriginalAliases($urlAliasesData);
+
+        if (count($originalUrlAliases) === count($urlAliasesData)) {
+            // no archived aliases - nothing to fix
+            return;
+        }
+
+        $updateQueryBuilder = $this->connection->createQueryBuilder();
+        $expr = $updateQueryBuilder->expr();
+        $updateQueryBuilder
+            ->update('ezurlalias_ml')
+            ->set('link', ':linkId')
+            ->set('parent', ':parentId')
+            ->where(
+                $expr->eq('action', ':action')
+            )
+            ->andWhere(
+                $expr->eq(
+                    'is_original',
+                    $updateQueryBuilder->createNamedParameter(0, \PDO::PARAM_INT)
```

Comment:
Should not we use `ParameterType::INTEGER`?

Reply:
> Should not we use `ParameterType::INTEGER`?

On newer branches yes, on 1.7 there might be older Doctrine used AFAIR.